pegen/grammar.py

```diff
@@ -382,7 +382,7 @@ def nullable_visit(self, rules: Dict[str, Rule]) -> bool:
         return False
 
 
-class RepeatWithSeparator(Repeat):
+class Gather(Repeat):
```

Comment:
Note: a few lines below, the `__str__()` method wasn't changed when you required the trailing `+` per @apalala's suggestion.